packages/toolpad-core/src/AppProvider/AppProvider.tsx

```diff
@@ -1,23 +1,52 @@
 import * as React from 'react';
-import { AppRouterCacheProvider } from '@mui/material-nextjs/v14-appRouter';
 import { ThemeProvider } from '@mui/material/styles';
 import CssBaseline from '@mui/material/CssBaseline';
-
 import type { Theme } from '@emotion/react';
 
+export interface Branding {
+  title?: string;
+  logo?: React.ReactNode;
+}
+
+export interface NavigationPageItem {
+  kind?: 'page';
+  title: string;
+  path?: string;
+  icon?: React.ReactNode;
+  children?: Navigation;
+}
+
+export interface NavigationSubheaderItem {
+  kind: 'header';
+  title: string;
+}
+
+export interface NavigationDividerItem {
+  kind: 'divider';
+}
+
+export type NavigationItem = NavigationPageItem | NavigationSubheaderItem | NavigationDividerItem;
+
+export type Navigation = NavigationItem[];
+
+export const BrandingContext = React.createContext<Branding | null>(null);
+
+export const NavigationContext = React.createContext<Navigation>([]);
+
 interface AppProviderProps {
   children: React.ReactNode;
   theme: Theme;
```

Comment:
Does it have to be mandatory? We could provide a default.
```suggestion
  theme?: Theme;
```

Reply:
If you use `create-toolpad-app` it comes with a default theme, but it's not built into the library.
We could add it as a default theme though and not have it set by `create-toolpad-app`, i guess that sounds good.

Reply:
We can have it created by `create-toolpad-app`, but that doesn't mean this property has to be mandatory?

Reply:
First options I can think of:

1. Users can provide separate themes for light and dark mode;
2. Theme can take a function with `mode: PaletteMode` as an argument.

I was trying to keep it simple for now thinking we can still make breaking changes afterwards.
I guess option 2 should be more flexible as in option 1 it would be harder to change parts of the theme conditionally based on the mode. But probably we should also allow just passing in a theme in a straightforward way.